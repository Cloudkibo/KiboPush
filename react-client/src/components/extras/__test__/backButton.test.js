import React from 'react'
import ReactDOM from 'react-dom'
import BackButton from './../backButton'
import checkPropTypes from 'check-prop-types'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

describe('Testing BackButton Component', () => {
  afterEach(cleanup)

  it('renders without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<BackButton onBack={() => {}} />, div)
  })

  it('should not through a warning for propTypes', () => {
    const expectedProps = {
      onBack: () => {}
    };
    const propsErr = checkPropTypes(BackButton.propTypes, expectedProps, 'props', BackButton.name);
    expect(propsErr).toBeUndefined();
  })


  it('matches snapshot', () => {
      const tree = renderer.create(<BackButton onBack={() => {}} />).toJSON()
      expect(tree).toMatchSnapshot()
  })
})
