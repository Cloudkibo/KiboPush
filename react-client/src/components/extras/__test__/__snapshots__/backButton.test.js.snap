// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BackButton Component matches snapshot 1`] = `
<div
  aria-expanded="true"
  className="m-dropdown m-dropdown--inline m-dropdown--arrow m-dropdown--align-right m-dropdown--align-push"
  style={
    Object {
      "bottom": "40px",
      "position": "fixed",
      "right": "60px",
    }
  }
>
  <button
    className="m-portlet__nav-link btn btn-lg btn-primary  m-btn m-btn--outline-2x m-btn--air m-btn--icon m-btn--icon-only m-btn--pill  m-dropdown__toggle"
    href="#/"
    onClick={[Function]}
  >
    <i
      className="la la-arrow-left"
    />
  </button>
</div>
`;
